website/docs/registry/providers/publishing.html.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: "registry"
+page_title: "Terraform Registry - Publishing Providers"
+sidebar_current: "docs-registry-provider-publishing"
+description: |-
+  Publishing Providers to the Terraform Registry
+---
+
+~> __Publishing Beta__<br>
+~> Welcome! Thanks for your interest participating in our Providers in the Registry beta! Paired with Terraform 0.13, our vision is to make it easier than ever to discover, distribute, and maintain your provider(s). We welcome any feedback you have throughout the process and encourage you to reach out if you have any questions or issues by emailing terraform-registry-beta@hashicorp.com.
+
+## Preparing your Provider
+
+### Writing a Provider
+
+Providers published to the Terraform Registry are written and built in the same way as other Terraform Providers. For guidance on how to write a provider, see [Writing Custom Providers](/docs/extend/writing-custom-providers.html).
+
+The provider repository on GitHub must match the pattern `terraform-provider-{NAME}`, and the repository must be public.
+
+### Documenting your Provider
+
+Your provider should contain an overview document (index.md), as well as a doc for each resource and data-source. See [Documenting Providers](documenting.html) for details about how to ensure your provider documentation renders properly on the Terraform Registry.
+
+### Creating a GitHub Release
```

Comment:
@justincampbell do we want to mention our best practices on pre-release versions? i.e. prerelease versions are delimited with a hyphen so to avoid them showing up in the registry?

Reply:
Yup! Good idea, I'll add that.

Reply:
Updated!

> Creating a GitHub Release
> 
> Publishing a provider requires at least one version be available on GitHub Releases. The tag must be a valid Semantic Version preceded with a v (for example, v1.2.3).
> 
> Terraform CLI and the Terraform Registry follow the Semantic Versioning specification when detecting a valid version, sorting versions, solving version constraints, and choosing the latest version. Prerelease versions are supported (available if explicitly defined but not chosen automatically) with a hyphen (-) delimiter, such as v1.2.3-pre.